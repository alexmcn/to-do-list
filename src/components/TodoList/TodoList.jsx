import React from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from '@material-ui/core/styles';
import { List } from '@material-ui/core';

const useStyles = makeStyles(theme => ({
  root: {
    width: '100%',
    backgroundColor: theme.palette.background.paper,
  },
}));

const TodoList = ({ todos }) => {
  const classes = useStyles();
  if (!todos || !todos.length) return null;

  return (
    <List className={classes.root}>
      <pre>
        {todos.map(todo => {
          return JSON.stringify(todo, null, ' ');
        })}
      </pre>
    </List>
  );
};

TodoList.propTypes = {
  todos: PropTypes.array,
};

export default TodoList;
