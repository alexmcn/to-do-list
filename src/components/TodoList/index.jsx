import React, { useContext } from 'react';
import { TodosContext } from 'context';
import Component from './TodoList';

export default () => {
  const { state } = useContext(TodosContext);

  return <Component todos={state.todos} />;
};
