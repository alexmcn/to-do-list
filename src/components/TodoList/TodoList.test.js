import React from 'react';
import renderer from 'react-test-renderer';

import TodoList from './TodoList';

describe('TodoList', () => {
  it('lists current todo list items if any exist', () => {
    const instance = renderer.create(
      <TodoList
        todos={[
          { id: 1, title: 'to do a', todoListId: '1', isDone: true },
          { id: 2, title: 'to do b', todoListId: '1', isDone: false },
          { id: 3, title: 'to do c', todoListId: '2', isDone: true },
        ]}
      />
    );

    expect(instance.toJSON()).toMatchSnapshot();
  });
});
