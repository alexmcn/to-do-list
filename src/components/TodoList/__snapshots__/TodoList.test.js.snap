// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList lists current todo list items if any exist 1`] = `
<ul
  className="MuiList-root makeStyles-root-1 MuiList-padding"
>
  <pre>
    {
 "id": 1,
 "title": "to do a",
 "todoListId": "1",
 "isDone": true
}
    {
 "id": 2,
 "title": "to do b",
 "todoListId": "1",
 "isDone": false
}
    {
 "id": 3,
 "title": "to do c",
 "todoListId": "2",
 "isDone": true
}
  </pre>
</ul>
`;
